src/applications/personalization/profile360/actions/updaters.js

```diff
@@ -55,7 +69,8 @@ export function refreshTransaction(transaction, analyticsSectionName) {
 
       if (isSuccessfulTransaction(transactionRefreshed)) {
         const forceCacheClear = true;
-        dispatch(refreshProfile(forceCacheClear));
+        await dispatch(refreshProfile(forceCacheClear));
+        dispatch(clearTransaction(transactionRefreshed));
         recordEvent({ event: 'profile-saved' });
```

Comment:
This is the main logical change - once transaction is updated to successful and the profile then finishes updating, we clear the transaction from state.